src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -174,6 +176,61 @@ _mongoc_write_command_init_insert (mongoc_write_command_t *command, /* IN */
 }
 
 
+// `_mongoc_write_command_init_insert_one_idl` returns the inserted ID in `inserted_id`.
+// Only called by mongoc_collection_insert_one.
+void
+_mongoc_write_command_init_insert_one_idl (mongoc_write_command_t *command,
+                                           const bson_t *document,
+                                           const bson_t *cmd_opts,
+                                           bson_t *insert_id,
+                                           int64_t operation_id)
+{
+   mongoc_bulk_write_flags_t flags = MONGOC_BULK_WRITE_FLAGS_INIT;
+
+   ENTRY;
+
+   BSON_ASSERT (command);
+
+   _mongoc_write_command_init_bulk (command, MONGOC_WRITE_COMMAND_INSERT, flags, operation_id, cmd_opts);
+   bson_init(insert_id);
+
+   /* must handle NULL document from mongoc_collection_insert_bulk */
+   /* near identical to _mongoc_write_command_insert_append but additionally records the inserted id */
+   if (document) {
```

Comment:
Suggest removing `if (document)` and associated comments. This function is only called by `mongoc_collection_insert_one`.